### DISCUSSION TITLE
how to "properly" blend

### DISCUSSION THREAD

**Discussion by assmikDuDuDuJ** [Sixze/ALS-Refactored#242] [Q&A] [answered]
hello , newbie question, how to "properly" blend/synchronize montages? I mean with overlay and locomotion animation, if I use blend in\out I get a foot sliding from overlay foot position, to montage position.
And after, if the animation ends on the right foot, the next step still starts from the right, there is no synchronization. 
I kind of poked around, but could not find how to fix it

**Comment by Sixze** [comment, marked as answer]
This is not an easy case, I can only point you where to look:

1. Learn how foot lock and dynamic transitions work to avoid foot sliding. In `AM_Als_Roll` you can see how curves can be used to lock feet in place when they touch the ground. Also, if you look at this animation in the game, you will see how dynamic transitions work, how they automatically and without sliding move the foot from the pose from the last frames of `AM_Als_Roll` to the pose from the overlay states.

2. Learn about Sync Groups, especially note that since UE 5.0 they can also be configured in animation montages. They should help keep movement animations in sync with animation montages so that movement starts on the correct foot.

**Reply by assmikDuDuDuJ** [reply]
Thank you for your response sir. 
The first thing I did was to look at how everything is done in ALS.
But it seems that I don't understand it very well.
In general, all the animations work as it should, but the transitions are a problem.
in my case, 5.0.1 Sync Groups in montages caused frequent crashes.

**Reply by Sixze** [reply]
Unfortunately I haven't tried using sync groups with animation montages yet, so I can't help with crashes or suggest anything else, but this is what I would try first if I ran into a issue like yours.
